smif/model/scenario_model.py

```diff
@@ -14,27 +13,75 @@ class ScenarioModel(Model):
     ---------
     name : string
         The unique name of this scenario
-    output : smif.metadata.MetaData
-        A name for the scenario output parameter
     """
 
-    def __init__(self, name, output=None):
-        if output:
-            if isinstance(output, MetadataSet):
-                super().__init__(name)
-                self._model_outputs = output
-            else:
-                msg = "output argument should be type smif.metadata.MetadataSet"
-                raise TypeError(msg)
-        else:
-            super().__init__(name)
+    def __init__(self, name):
+        super().__init__(name)
 
         self._data = {}
         self.timesteps = []
 
-    @property
-    def data(self):
-        return self._data
+        "The scenario set to which this scenario belongs"
+        self.scenario_set = None
+        self._filename = {}
+
+    def as_dict(self):
+
+        config = {
+            'name': self.name,
+            'description': self.description,
+            'scenario_set': self.scenario_set}
+
+        parameters = []
+        for output in self.model_outputs:
+
+            parameters.append({
+                'name': output.name,
+                'filename': self.get_filename(output.name),
+                'spatial_resolution': output.spatial_resolution.name,
+                'temporal_resolution': output.temporal_resolution.name,
+                'units': output.units
+                })
+
+        config['parameters'] = parameters
+
+        return config
+
+    def set_filename(self, output, filename):
+        """Store the name of the data file associate with `output`
+
+        Arguments
+        ---------
+        output : str
+        filename : str
+        """
+        self._filename[output] = filename
+
+    def get_filename(self, output):
+        """Get the name of the datafile associated with `output`
+
+        Arguments
+        ---------
+        output : str
+
+        Returns
+        -------
+        str
+        """
+        self._check_output(output)
+        return self._filename[output]
+
+    def get_data(self, output):
```

Comment:
Could `get_data` go to the data layer to retrieve scenario data? Aiming to avoid holding details of filenames (or other data access metadata) per-parameter.

Reply:
Not sure where it would go in data_layer at the moment?  DatafileInterface reads/writes dicts which includes filenames.